In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/weather-dataset-rattle-package/weatherAUS.csv


1. Read the dataset as a dataframe. Create a copy of your dataframe. Solve the rest of the questions using this dataframe copy.

In [2]:
a = pd.read_csv("../input/weather-dataset-rattle-package/weatherAUS.csv")
a

Date Location  MinTemp  MaxTemp  Rainfall  Evaporation  \
0       2008-12-01   Albury     13.4     22.9       0.6          NaN   
1       2008-12-02   Albury      7.4     25.1       0.0          NaN   
2       2008-12-03   Albury     12.9     25.7       0.0          NaN   
3       2008-12-04   Albury      9.2     28.0       0.0          NaN   
4       2008-12-05   Albury     17.5     32.3       1.0          NaN   
...            ...      ...      ...      ...       ...          ...   
145455  2017-06-21    Uluru      2.8     23.4       0.0          NaN   
145456  2017-06-22    Uluru      3.6     25.3       0.0          NaN   
145457  2017-06-23    Uluru      5.4     26.9       0.0          NaN   
145458  2017-06-24    Uluru      7.8     27.0       0.0          NaN   
145459  2017-06-25    Uluru     14.9      NaN       0.0          NaN   

        Sunshine WindGustDir  WindGustSpeed WindDir9am  ... Humidity9am  \
0            NaN           W           44.0          W  ...        71.0   
1            NaN         WNW           44.0        NNW  ...        44.0   
2            NaN         WSW           46.0          W  ...        38.0   
3            NaN          NE           24.0         SE  ...        45.0   
4            NaN           W           41.0        ENE  ...        82.0   
...          ...         ...            ...        ...  ...         ...   
145455       NaN           E           31.0         SE  ...        51.0   
145456       NaN         NNW           22.0         SE  ...        56.0   
145457       NaN           N           37.0         SE  ...        53.0   
145458       NaN          SE           28.0        SSE  ...        51.0   
145459       NaN         NaN            NaN        ESE  ...        62.0   

        Humidity3pm  Pressure9am  Pressure3pm  Cloud9am  Cloud3pm  Temp9am  \
0              22.0       1007.7       1007.1       8.0       NaN     16.9   
1              25.0       1010.6       1007.8       NaN       NaN     17.2   
2              30.0       1007.6       1008.7       NaN       2.0     21.0   
3              16.0       1017.6       1012.8       NaN       NaN     18.1   
4              33.0       1010.8       1006.0       7.0       8.0     17.8   
...             ...          ...          ...       ...       ...      ...   
145455         24.0       1024.6       1020.3       NaN       NaN     10.1   
145456         21.0       1023.5       1019.1       NaN       NaN     10.9   
145457         24.0       1021.0       1016.8       NaN       NaN     12.5   
145458         24.0       1019.4       1016.5       3.0       2.0     15.1   
145459         36.0       1020.2       1017.9       8.0       8.0     15.0   

        Temp3pm  RainToday  RainTomorrow  
0          21.8         No            No  
1          24.3         No            No  
2          23.2         No            No  
3          26.5         No            No  
4          29.7         No            No  
...         ...        ...           ...  
145455     22.4         No            No  
145456     24.5         No            No  
145457     26.1         No            No  
145458     26.0         No            No  
145459     20.9         No           NaN  

[145460 rows x 23 columns]

In [3]:
df = a.copy()

In [4]:
df.head()

Date Location  MinTemp  MaxTemp  Rainfall  Evaporation  Sunshine  \
0  2008-12-01   Albury     13.4     22.9       0.6          NaN       NaN   
1  2008-12-02   Albury      7.4     25.1       0.0          NaN       NaN   
2  2008-12-03   Albury     12.9     25.7       0.0          NaN       NaN   
3  2008-12-04   Albury      9.2     28.0       0.0          NaN       NaN   
4  2008-12-05   Albury     17.5     32.3       1.0          NaN       NaN   

  WindGustDir  WindGustSpeed WindDir9am  ... Humidity9am  Humidity3pm  \
0           W           44.0          W  ...        71.0         22.0   
1         WNW           44.0        NNW  ...        44.0         25.0   
2         WSW           46.0          W  ...        38.0         30.0   
3          NE           24.0         SE  ...        45.0         16.0   
4           W           41.0        ENE  ...        82.0         33.0   

   Pressure9am  Pressure3pm  Cloud9am  Cloud3pm  Temp9am  Temp3pm  RainToday  \
0       1007.7       1007.1       8.0       NaN     16.9     21.8         No   
1       1010.6       1007.8       NaN       NaN     17.2     24.3         No   
2       1007.6       1008.7       NaN       2.0     21.0     23.2         No   
3       1017.6       1012.8       NaN       NaN     18.1     26.5         No   
4       1010.8       1006.0       7.0       8.0     17.8     29.7         No   

   RainTomorrow  
0            No  
1            No  
2            No  
3            No  
4            No  

[5 rows x 23 columns]

2. Find out the describe and info attributes of the dataframe. Analyze these information and create a short write-up according to your findings.

In [5]:
df.describe().T

count         mean        std    min     25%     50%  \
MinTemp        143975.0    12.194034   6.398495   -8.5     7.6    12.0   
MaxTemp        144199.0    23.221348   7.119049   -4.8    17.9    22.6   
Rainfall       142199.0     2.360918   8.478060    0.0     0.0     0.0   
Evaporation     82670.0     5.468232   4.193704    0.0     2.6     4.8   
Sunshine        75625.0     7.611178   3.785483    0.0     4.8     8.4   
WindGustSpeed  135197.0    40.035230  13.607062    6.0    31.0    39.0   
WindSpeed9am   143693.0    14.043426   8.915375    0.0     7.0    13.0   
WindSpeed3pm   142398.0    18.662657   8.809800    0.0    13.0    19.0   
Humidity9am    142806.0    68.880831  19.029164    0.0    57.0    70.0   
Humidity3pm    140953.0    51.539116  20.795902    0.0    37.0    52.0   
Pressure9am    130395.0  1017.649940   7.106530  980.5  1012.9  1017.6   
Pressure3pm    130432.0  1015.255889   7.037414  977.1  1010.4  1015.2   
Cloud9am        89572.0     4.447461   2.887159    0.0     1.0     5.0   
Cloud3pm        86102.0     4.509930   2.720357    0.0     2.0     5.0   
Temp9am        143693.0    16.990631   6.488753   -7.2    12.3    16.7   
Temp3pm        141851.0    21.683390   6.936650   -5.4    16.6    21.1   

                  75%     max  
MinTemp          16.9    33.9  
MaxTemp          28.2    48.1  
Rainfall          0.8   371.0  
Evaporation       7.4   145.0  
Sunshine         10.6    14.5  
WindGustSpeed    48.0   135.0  
WindSpeed9am     19.0   130.0  
WindSpeed3pm     24.0    87.0  
Humidity9am      83.0   100.0  
Humidity3pm      66.0   100.0  
Pressure9am    1022.4  1041.0  
Pressure3pm    1020.0  1039.6  
Cloud9am          7.0     9.0  
Cloud3pm          7.0     9.0  
Temp9am          21.6    40.2  
Temp3pm          26.4    46.7

Coment; Bu tabloda Avustralya'nın min. sıcaklık, max. sıcaklık, ruzgar hızı, nem, basinc vs. hava durumu ile ilgili bilgilerini görebiliriz. Ornegin min. sıcaklık -8.5 ve max. sıcaklık 48.1 derecedir.

Ayrıca mean ve %50 column degerlerine baktigimizda bunlarin birbirlerine yakin degerler olduklarini da goruruz. Bu da bize bu veri setinin normal bir dagilim gosterdigini ifade eder.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145460 entries, 0 to 145459
Data columns (total 23 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   Date           145460 non-null  object 
 1   Location       145460 non-null  object 
 2   MinTemp        143975 non-null  float64
 3   MaxTemp        144199 non-null  float64
 4   Rainfall       142199 non-null  float64
 5   Evaporation    82670 non-null   float64
 6   Sunshine       75625 non-null   float64
 7   WindGustDir    135134 non-null  object 
 8   WindGustSpeed  135197 non-null  float64
 9   WindDir9am     134894 non-null  object 
 10  WindDir3pm     141232 non-null  object 
 11  WindSpeed9am   143693 non-null  float64
 12  WindSpeed3pm   142398 non-null  float64
 13  Humidity9am    142806 non-null  float64
 14  Humidity3pm    140953 non-null  float64
 15  Pressure9am    130395 non-null  float64
 16  Pressure3pm    130432 non-null  float64
 17  Cloud9am       89572 non-null

Coment; df.info ile data tiplerini(ornegin yukarida float64(16), object(7)), kac tane column oldugunu(23) ve bunlarin isimlerini ve ne kadar non-null deger oldugunu gorebiliriz.

3. Find out the shape and size info of the dataset.

In [7]:
df.shape

(145460, 23)

In [8]:
df.size

3345580

4. Find out the types values of the columns and save the result as a dataframe.

In [9]:
b = df.dtypes.to_frame("dtypes").reset_index()
b

index   dtypes
0            Date   object
1        Location   object
2         MinTemp  float64
3         MaxTemp  float64
4        Rainfall  float64
5     Evaporation  float64
6        Sunshine  float64
7     WindGustDir   object
8   WindGustSpeed  float64
9      WindDir9am   object
10     WindDir3pm   object
11   WindSpeed9am  float64
12   WindSpeed3pm  float64
13    Humidity9am  float64
14    Humidity3pm  float64
15    Pressure9am  float64
16    Pressure3pm  float64
17       Cloud9am  float64
18       Cloud3pm  float64
19        Temp9am  float64
20        Temp3pm  float64
21      RainToday   object
22   RainTomorrow   object

In [10]:
dtypes = pd.DataFrame(df.dtypes,columns = ["dtypes"])
dtypes

dtypes
Date            object
Location        object
MinTemp        float64
MaxTemp        float64
Rainfall       float64
Evaporation    float64
Sunshine       float64
WindGustDir     object
WindGustSpeed  float64
WindDir9am      object
WindDir3pm      object
WindSpeed9am   float64
WindSpeed3pm   float64
Humidity9am    float64
Humidity3pm    float64
Pressure9am    float64
Pressure3pm    float64
Cloud9am       float64
Cloud3pm       float64
Temp9am        float64
Temp3pm        float64
RainToday       object
RainTomorrow    object

5. Find out the non-null counts of the columns and save the result as a dataframe.

In [11]:
not_null = pd.DataFrame(df.notnull().sum())
not_null

0
Date           145460
Location       145460
MinTemp        143975
MaxTemp        144199
Rainfall       142199
Evaporation     82670
Sunshine        75625
WindGustDir    135134
WindGustSpeed  135197
WindDir9am     134894
WindDir3pm     141232
WindSpeed9am   143693
WindSpeed3pm   142398
Humidity9am    142806
Humidity3pm    140953
Pressure9am    130395
Pressure3pm    130432
Cloud9am        89572
Cloud3pm        86102
Temp9am        143693
Temp3pm        141851
RainToday      142199
RainTomorrow   142193

6. Find out the null counts of the columns and save the result as a dataframe.

In [12]:
non_null = pd.DataFrame(df.isnull().sum(), columns=["is_null"])
non_null

is_null
Date                 0
Location             0
MinTemp           1485
MaxTemp           1261
Rainfall          3261
Evaporation      62790
Sunshine         69835
WindGustDir      10326
WindGustSpeed    10263
WindDir9am       10566
WindDir3pm        4228
WindSpeed9am      1767
WindSpeed3pm      3062
Humidity9am       2654
Humidity3pm       4507
Pressure9am      15065
Pressure3pm      15028
Cloud9am         55888
Cloud3pm         59358
Temp9am           1767
Temp3pm           3609
RainToday         3261
RainTomorrow      3267

7. Find out the unique counts of the columns and save the result as a dataframe.

In [13]:
nunique = df.nunique().to_frame("nunique").reset_index()
nunique

index  nunique
0            Date     3436
1        Location       49
2         MinTemp      389
3         MaxTemp      505
4        Rainfall      681
5     Evaporation      358
6        Sunshine      145
7     WindGustDir       16
8   WindGustSpeed       67
9      WindDir9am       16
10     WindDir3pm       16
11   WindSpeed9am       43
12   WindSpeed3pm       44
13    Humidity9am      101
14    Humidity3pm      101
15    Pressure9am      546
16    Pressure3pm      549
17       Cloud9am       10
18       Cloud3pm       10
19        Temp9am      441
20        Temp3pm      502
21      RainToday        2
22   RainTomorrow        2

II. YOL

In [14]:
nunique = pd.DataFrame(df.nunique(), columns = ["nunique"])
nunique

nunique
Date              3436
Location            49
MinTemp            389
MaxTemp            505
Rainfall           681
Evaporation        358
Sunshine           145
WindGustDir         16
WindGustSpeed       67
WindDir9am          16
WindDir3pm          16
WindSpeed9am        43
WindSpeed3pm        44
Humidity9am        101
Humidity3pm        101
Pressure9am        546
Pressure3pm        549
Cloud9am            10
Cloud3pm            10
Temp9am            441
Temp3pm            502
RainToday            2
RainTomorrow         2

8. Merge the dataframes you created in questions 4-5-6-7.

In [15]:
c = pd.concat([dtypes,not_null,non_null, nunique], axis=1)
c

dtypes       0  is_null  nunique
Date            object  145460        0     3436
Location        object  145460        0       49
MinTemp        float64  143975     1485      389
MaxTemp        float64  144199     1261      505
Rainfall       float64  142199     3261      681
Evaporation    float64   82670    62790      358
Sunshine       float64   75625    69835      145
WindGustDir     object  135134    10326       16
WindGustSpeed  float64  135197    10263       67
WindDir9am      object  134894    10566       16
WindDir3pm      object  141232     4228       16
WindSpeed9am   float64  143693     1767       43
WindSpeed3pm   float64  142398     3062       44
Humidity9am    float64  142806     2654      101
Humidity3pm    float64  140953     4507      101
Pressure9am    float64  130395    15065      546
Pressure3pm    float64  130432    15028      549
Cloud9am       float64   89572    55888       10
Cloud3pm       float64   86102    59358       10
Temp9am        float64  143693     1767      441
Temp3pm        float64  141851     3609      502
RainToday       object  142199     3261        2
RainTomorrow    object  142193     3267        2

9. Lowercase all column names.

In [16]:
df.columns = df.columns.str.lower()
df.head()

date location  mintemp  maxtemp  rainfall  evaporation  sunshine  \
0  2008-12-01   Albury     13.4     22.9       0.6          NaN       NaN   
1  2008-12-02   Albury      7.4     25.1       0.0          NaN       NaN   
2  2008-12-03   Albury     12.9     25.7       0.0          NaN       NaN   
3  2008-12-04   Albury      9.2     28.0       0.0          NaN       NaN   
4  2008-12-05   Albury     17.5     32.3       1.0          NaN       NaN   

  windgustdir  windgustspeed winddir9am  ... humidity9am  humidity3pm  \
0           W           44.0          W  ...        71.0         22.0   
1         WNW           44.0        NNW  ...        44.0         25.0   
2         WSW           46.0          W  ...        38.0         30.0   
3          NE           24.0         SE  ...        45.0         16.0   
4           W           41.0        ENE  ...        82.0         33.0   

   pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  raintoday  \
0       1007.7       1007.1       8.0       NaN     16.9     21.8         No   
1       1010.6       1007.8       NaN       NaN     17.2     24.3         No   
2       1007.6       1008.7       NaN       2.0     21.0     23.2         No   
3       1017.6       1012.8       NaN       NaN     18.1     26.5         No   
4       1010.8       1006.0       7.0       8.0     17.8     29.7         No   

   raintomorrow  
0            No  
1            No  
2            No  
3            No  
4            No  

[5 rows x 23 columns]

10. Change all the No values to NoRain and all the Yes values to Rain in raintoday and raintomorrow columns.

In [17]:
df.raintoday.replace('No','NoRain',inplace = True)
df.raintoday.replace('Yes','Rain',inplace = True)
df.raintomorrow.replace('No','NoRain',inplace = True)
df.raintomorrow.replace('Yes','Rain',inplace = True)
df.head(10)

date location  mintemp  maxtemp  rainfall  evaporation  sunshine  \
0  2008-12-01   Albury     13.4     22.9       0.6          NaN       NaN   
1  2008-12-02   Albury      7.4     25.1       0.0          NaN       NaN   
2  2008-12-03   Albury     12.9     25.7       0.0          NaN       NaN   
3  2008-12-04   Albury      9.2     28.0       0.0          NaN       NaN   
4  2008-12-05   Albury     17.5     32.3       1.0          NaN       NaN   
5  2008-12-06   Albury     14.6     29.7       0.2          NaN       NaN   
6  2008-12-07   Albury     14.3     25.0       0.0          NaN       NaN   
7  2008-12-08   Albury      7.7     26.7       0.0          NaN       NaN   
8  2008-12-09   Albury      9.7     31.9       0.0          NaN       NaN   
9  2008-12-10   Albury     13.1     30.1       1.4          NaN       NaN   

  windgustdir  windgustspeed winddir9am  ... humidity9am  humidity3pm  \
0           W           44.0          W  ...        71.0         22.0   
1         WNW           44.0        NNW  ...        44.0         25.0   
2         WSW           46.0          W  ...        38.0         30.0   
3          NE           24.0         SE  ...        45.0         16.0   
4           W           41.0        ENE  ...        82.0         33.0   
5         WNW           56.0          W  ...        55.0         23.0   
6           W           50.0         SW  ...        49.0         19.0   
7           W           35.0        SSE  ...        48.0         19.0   
8         NNW           80.0         SE  ...        42.0          9.0   
9           W           28.0          S  ...        58.0         27.0   

   pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  raintoday  \
0       1007.7       1007.1       8.0       NaN     16.9     21.8     NoRain   
1       1010.6       1007.8       NaN       NaN     17.2     24.3     NoRain   
2       1007.6       1008.7       NaN       2.0     21.0     23.2     NoRain   
3       1017.6       1012.8       NaN       NaN     18.1     26.5     NoRain   
4       1010.8       1006.0       7.0       8.0     17.8     29.7     NoRain   
5       1009.2       1005.4       NaN       NaN     20.6     28.9     NoRain   
6       1009.6       1008.2       1.0       NaN     18.1     24.6     NoRain   
7       1013.4       1010.1       NaN       NaN     16.3     25.5     NoRain   
8       1008.9       1003.6       NaN       NaN     18.3     30.2     NoRain   
9       1007.0       1005.7       NaN       NaN     20.1     28.2       Rain   

   raintomorrow  
0        NoRain  
1        NoRain  
2        NoRain  
3        NoRain  
4        NoRain  
5        NoRain  
6        NoRain  
7        NoRain  
8          Rain  
9        NoRain  

[10 rows x 23 columns]

11. Change the data type of "date" (object) column to datetime64 and reformat the date as DD/MM/YYYY.

In [18]:
df.date = pd.to_datetime(df.date)
df.date = df.date.dt.strftime('%d/%m/%Y')
df.head()

date location  mintemp  maxtemp  rainfall  evaporation  sunshine  \
0  01/12/2008   Albury     13.4     22.9       0.6          NaN       NaN   
1  02/12/2008   Albury      7.4     25.1       0.0          NaN       NaN   
2  03/12/2008   Albury     12.9     25.7       0.0          NaN       NaN   
3  04/12/2008   Albury      9.2     28.0       0.0          NaN       NaN   
4  05/12/2008   Albury     17.5     32.3       1.0          NaN       NaN   

  windgustdir  windgustspeed winddir9am  ... humidity9am  humidity3pm  \
0           W           44.0          W  ...        71.0         22.0   
1         WNW           44.0        NNW  ...        44.0         25.0   
2         WSW           46.0          W  ...        38.0         30.0   
3          NE           24.0         SE  ...        45.0         16.0   
4           W           41.0        ENE  ...        82.0         33.0   

   pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  raintoday  \
0       1007.7       1007.1       8.0       NaN     16.9     21.8     NoRain   
1       1010.6       1007.8       NaN       NaN     17.2     24.3     NoRain   
2       1007.6       1008.7       NaN       2.0     21.0     23.2     NoRain   
3       1017.6       1012.8       NaN       NaN     18.1     26.5     NoRain   
4       1010.8       1006.0       7.0       8.0     17.8     29.7     NoRain   

   raintomorrow  
0        NoRain  
1        NoRain  
2        NoRain  
3        NoRain  
4        NoRain  

[5 rows x 23 columns]

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145460 entries, 0 to 145459
Data columns (total 23 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   date           145460 non-null  object 
 1   location       145460 non-null  object 
 2   mintemp        143975 non-null  float64
 3   maxtemp        144199 non-null  float64
 4   rainfall       142199 non-null  float64
 5   evaporation    82670 non-null   float64
 6   sunshine       75625 non-null   float64
 7   windgustdir    135134 non-null  object 
 8   windgustspeed  135197 non-null  float64
 9   winddir9am     134894 non-null  object 
 10  winddir3pm     141232 non-null  object 
 11  windspeed9am   143693 non-null  float64
 12  windspeed3pm   142398 non-null  float64
 13  humidity9am    142806 non-null  float64
 14  humidity3pm    140953 non-null  float64
 15  pressure9am    130395 non-null  float64
 16  pressure3pm    130432 non-null  float64
 17  cloud9am       89572 non-null

12. Create a new column called "difference", calculate the difference between maxtemp and mintemp columns for each row, and store the value in this new column.

In [20]:
df

date location  mintemp  maxtemp  rainfall  evaporation  \
0       01/12/2008   Albury     13.4     22.9       0.6          NaN   
1       02/12/2008   Albury      7.4     25.1       0.0          NaN   
2       03/12/2008   Albury     12.9     25.7       0.0          NaN   
3       04/12/2008   Albury      9.2     28.0       0.0          NaN   
4       05/12/2008   Albury     17.5     32.3       1.0          NaN   
...            ...      ...      ...      ...       ...          ...   
145455  21/06/2017    Uluru      2.8     23.4       0.0          NaN   
145456  22/06/2017    Uluru      3.6     25.3       0.0          NaN   
145457  23/06/2017    Uluru      5.4     26.9       0.0          NaN   
145458  24/06/2017    Uluru      7.8     27.0       0.0          NaN   
145459  25/06/2017    Uluru     14.9      NaN       0.0          NaN   

        sunshine windgustdir  windgustspeed winddir9am  ... humidity9am  \
0            NaN           W           44.0          W  ...        71.0   
1            NaN         WNW           44.0        NNW  ...        44.0   
2            NaN         WSW           46.0          W  ...        38.0   
3            NaN          NE           24.0         SE  ...        45.0   
4            NaN           W           41.0        ENE  ...        82.0   
...          ...         ...            ...        ...  ...         ...   
145455       NaN           E           31.0         SE  ...        51.0   
145456       NaN         NNW           22.0         SE  ...        56.0   
145457       NaN           N           37.0         SE  ...        53.0   
145458       NaN          SE           28.0        SSE  ...        51.0   
145459       NaN         NaN            NaN        ESE  ...        62.0   

        humidity3pm  pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  \
0              22.0       1007.7       1007.1       8.0       NaN     16.9   
1              25.0       1010.6       1007.8       NaN       NaN     17.2   
2              30.0       1007.6       1008.7       NaN       2.0     21.0   
3              16.0       1017.6       1012.8       NaN       NaN     18.1   
4              33.0       1010.8       1006.0       7.0       8.0     17.8   
...             ...          ...          ...       ...       ...      ...   
145455         24.0       1024.6       1020.3       NaN       NaN     10.1   
145456         21.0       1023.5       1019.1       NaN       NaN     10.9   
145457         24.0       1021.0       1016.8       NaN       NaN     12.5   
145458         24.0       1019.4       1016.5       3.0       2.0     15.1   
145459         36.0       1020.2       1017.9       8.0       8.0     15.0   

        temp3pm  raintoday  raintomorrow  
0          21.8     NoRain        NoRain  
1          24.3     NoRain        NoRain  
2          23.2     NoRain        NoRain  
3          26.5     NoRain        NoRain  
4          29.7     NoRain        NoRain  
...         ...        ...           ...  
145455     22.4     NoRain        NoRain  
145456     24.5     NoRain        NoRain  
145457     26.1     NoRain        NoRain  
145458     26.0     NoRain        NoRain  
145459     20.9     NoRain           NaN  

[145460 rows x 23 columns]

In [21]:
df["difference"] = df["maxtemp"] - df["mintemp"]
df

date location  mintemp  maxtemp  rainfall  evaporation  \
0       01/12/2008   Albury     13.4     22.9       0.6          NaN   
1       02/12/2008   Albury      7.4     25.1       0.0          NaN   
2       03/12/2008   Albury     12.9     25.7       0.0          NaN   
3       04/12/2008   Albury      9.2     28.0       0.0          NaN   
4       05/12/2008   Albury     17.5     32.3       1.0          NaN   
...            ...      ...      ...      ...       ...          ...   
145455  21/06/2017    Uluru      2.8     23.4       0.0          NaN   
145456  22/06/2017    Uluru      3.6     25.3       0.0          NaN   
145457  23/06/2017    Uluru      5.4     26.9       0.0          NaN   
145458  24/06/2017    Uluru      7.8     27.0       0.0          NaN   
145459  25/06/2017    Uluru     14.9      NaN       0.0          NaN   

        sunshine windgustdir  windgustspeed winddir9am  ... humidity3pm  \
0            NaN           W           44.0          W  ...        22.0   
1            NaN         WNW           44.0        NNW  ...        25.0   
2            NaN         WSW           46.0          W  ...        30.0   
3            NaN          NE           24.0         SE  ...        16.0   
4            NaN           W           41.0        ENE  ...        33.0   
...          ...         ...            ...        ...  ...         ...   
145455       NaN           E           31.0         SE  ...        24.0   
145456       NaN         NNW           22.0         SE  ...        21.0   
145457       NaN           N           37.0         SE  ...        24.0   
145458       NaN          SE           28.0        SSE  ...        24.0   
145459       NaN         NaN            NaN        ESE  ...        36.0   

        pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  \
0            1007.7       1007.1       8.0       NaN     16.9     21.8   
1            1010.6       1007.8       NaN       NaN     17.2     24.3   
2            1007.6       1008.7       NaN       2.0     21.0     23.2   
3            1017.6       1012.8       NaN       NaN     18.1     26.5   
4            1010.8       1006.0       7.0       8.0     17.8     29.7   
...             ...          ...       ...       ...      ...      ...   
145455       1024.6       1020.3       NaN       NaN     10.1     22.4   
145456       1023.5       1019.1       NaN       NaN     10.9     24.5   
145457       1021.0       1016.8       NaN       NaN     12.5     26.1   
145458       1019.4       1016.5       3.0       2.0     15.1     26.0   
145459       1020.2       1017.9       8.0       8.0     15.0     20.9   

        raintoday  raintomorrow  difference  
0          NoRain        NoRain         9.5  
1          NoRain        NoRain        17.7  
2          NoRain        NoRain        12.8  
3          NoRain        NoRain        18.8  
4          NoRain        NoRain        14.8  
...           ...           ...         ...  
145455     NoRain        NoRain        20.6  
145456     NoRain        NoRain        21.7  
145457     NoRain        NoRain        21.5  
145458     NoRain        NoRain        19.2  
145459     NoRain           NaN         NaN  

[145460 rows x 24 columns]

13. Remove the evaporation and sunshine columns from the dataset permanently.

In [22]:
df.drop("evaporation",axis = 1, inplace = True)
df.drop("sunshine",axis = 1, inplace = True)
df

date location  mintemp  maxtemp  rainfall windgustdir  \
0       01/12/2008   Albury     13.4     22.9       0.6           W   
1       02/12/2008   Albury      7.4     25.1       0.0         WNW   
2       03/12/2008   Albury     12.9     25.7       0.0         WSW   
3       04/12/2008   Albury      9.2     28.0       0.0          NE   
4       05/12/2008   Albury     17.5     32.3       1.0           W   
...            ...      ...      ...      ...       ...         ...   
145455  21/06/2017    Uluru      2.8     23.4       0.0           E   
145456  22/06/2017    Uluru      3.6     25.3       0.0         NNW   
145457  23/06/2017    Uluru      5.4     26.9       0.0           N   
145458  24/06/2017    Uluru      7.8     27.0       0.0          SE   
145459  25/06/2017    Uluru     14.9      NaN       0.0         NaN   

        windgustspeed winddir9am winddir3pm  windspeed9am  ...  humidity3pm  \
0                44.0          W        WNW          20.0  ...         22.0   
1                44.0        NNW        WSW           4.0  ...         25.0   
2                46.0          W        WSW          19.0  ...         30.0   
3                24.0         SE          E          11.0  ...         16.0   
4                41.0        ENE         NW           7.0  ...         33.0   
...               ...        ...        ...           ...  ...          ...   
145455           31.0         SE        ENE          13.0  ...         24.0   
145456           22.0         SE          N          13.0  ...         21.0   
145457           37.0         SE        WNW           9.0  ...         24.0   
145458           28.0        SSE          N          13.0  ...         24.0   
145459            NaN        ESE        ESE          17.0  ...         36.0   

        pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  \
0            1007.7       1007.1       8.0       NaN     16.9     21.8   
1            1010.6       1007.8       NaN       NaN     17.2     24.3   
2            1007.6       1008.7       NaN       2.0     21.0     23.2   
3            1017.6       1012.8       NaN       NaN     18.1     26.5   
4            1010.8       1006.0       7.0       8.0     17.8     29.7   
...             ...          ...       ...       ...      ...      ...   
145455       1024.6       1020.3       NaN       NaN     10.1     22.4   
145456       1023.5       1019.1       NaN       NaN     10.9     24.5   
145457       1021.0       1016.8       NaN       NaN     12.5     26.1   
145458       1019.4       1016.5       3.0       2.0     15.1     26.0   
145459       1020.2       1017.9       8.0       8.0     15.0     20.9   

        raintoday  raintomorrow difference  
0          NoRain        NoRain        9.5  
1          NoRain        NoRain       17.7  
2          NoRain        NoRain       12.8  
3          NoRain        NoRain       18.8  
4          NoRain        NoRain       14.8  
...           ...           ...        ...  
145455     NoRain        NoRain       20.6  
145456     NoRain        NoRain       21.7  
145457     NoRain        NoRain       21.5  
145458     NoRain        NoRain       19.2  
145459     NoRain           NaN        NaN  

[145460 rows x 22 columns]

14. Find out the most rainy day for each city.

In [23]:
df_most_rainy_day = df.groupby(df.location)["rainfall"].idxmax().to_frame("mostrainydays")
day = df_most_rainy_day.mostrainydays.to_list()
df.date[day]

98285     14/02/2014
109820    02/06/2012
1156      01/03/2012
136525    28/02/2010
4468      29/01/2013
55876     14/01/2011
58938     05/02/2011
86414     02/05/2015
87242     12/01/2009
46715     03/12/2010
7832      16/02/2014
9368      07/11/2009
81741     14/01/2011
140071    16/02/2011
91455     29/04/2012
132137    14/01/2015
143335    27/12/2015
135557    29/01/2016
68151     07/02/2011
65714     01/06/2013
71158     05/02/2011
13096     26/11/2011
100287    14/01/2011
53221     04/03/2012
17321     23/04/2015
73505     12/06/2013
19661     24/06/2013
23667     22/03/2016
103259    08/12/2010
115660    07/12/2011
25263     10/02/2012
123695    10/02/2017
118185    11/07/2010
75723     14/01/2011
29379     21/04/2015
63259     07/12/2014
125255    02/03/2013
32723     21/04/2015
35544     15/10/2014
93344     03/02/2009
49761     09/12/2010
145278    26/12/2016
36960     08/03/2010
128046    01/06/2012
79512     01/06/2013
42010     06/01/2016
113178    15/07/2013
44916     25/

15. Filter out all the data for the city 'Albury' and then sort according to maxtemp column.

In [24]:
df_Albury = df[df.location == "Albury"]
df_Albury

date location  mintemp  maxtemp  rainfall windgustdir  \
0     01/12/2008   Albury     13.4     22.9       0.6           W   
1     02/12/2008   Albury      7.4     25.1       0.0         WNW   
2     03/12/2008   Albury     12.9     25.7       0.0         WSW   
3     04/12/2008   Albury      9.2     28.0       0.0          NE   
4     05/12/2008   Albury     17.5     32.3       1.0           W   
...          ...      ...      ...      ...       ...         ...   
3035  21/06/2017   Albury      1.2     15.2       0.4         ENE   
3036  22/06/2017   Albury      0.8     13.4       0.0           W   
3037  23/06/2017   Albury      1.1     11.9       0.0          SE   
3038  24/06/2017   Albury      1.1     14.1       0.2         WSW   
3039  25/06/2017   Albury      3.9     10.9       0.0         NNW   

      windgustspeed winddir9am winddir3pm  windspeed9am  ...  humidity3pm  \
0              44.0          W        WNW          20.0  ...         22.0   
1              44.0        NNW        WSW           4.0  ...         25.0   
2              46.0          W        WSW          19.0  ...         30.0   
3              24.0         SE          E          11.0  ...         16.0   
4              41.0        ENE         NW           7.0  ...         33.0   
...             ...        ...        ...           ...  ...          ...   
3035           15.0        NaN        NNE           0.0  ...         62.0   
3036           17.0          S        NaN           6.0  ...         66.0   
3037           44.0        SSE        SSE           9.0  ...         81.0   
3038           28.0         SW          W           4.0  ...         49.0   
3039           28.0          S        NaN           6.0  ...         82.0   

      pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  \
0          1007.7       1007.1       8.0       NaN     16.9     21.8   
1          1010.6       1007.8       NaN       NaN     17.2     24.3   
2          1007.6       1008.7       NaN       2.0     21.0     23.2   
3          1017.6       1012.8       NaN       NaN     18.1     26.5   
4          1010.8       1006.0       7.0       8.0     17.8     29.7   
...           ...          ...       ...       ...      ...      ...   
3035       1029.4       1026.7       8.0       NaN      2.9     14.3   
3036       1029.4       1025.9       8.0       1.0      3.6     13.3   
3037       1022.3       1017.7       8.0       1.0      2.7     10.2   
3038       1018.8       1017.2       7.0       6.0      3.9     13.1   
3039       1020.5       1018.8       7.0       8.0      6.4      8.8   

      raintoday  raintomorrow difference  
0        NoRain        NoRain        9.5  
1        NoRain        NoRain       17.7  
2        NoRain        NoRain       12.8  
3        NoRain        NoRain       18.8  
4        NoRain        NoRain       14.8  
...         ...           ...        ...  
3035     NoRain        NoRain       14.0  
3036     NoRain        NoRain       12.6  
3037     NoRain        NoRain       10.8  
3038     NoRain        NoRain       13.0  
3039     NoRain          Rain        7.0  

[3040 rows x 22 columns]

In [25]:
df_Albury.sort_values(by = "maxtemp")

date location  mintemp  maxtemp  rainfall windgustdir  \
2328  15/07/2015   Albury      4.4      6.8      10.0          NE   
2675  26/06/2016   Albury     -1.3      7.5       0.0         ENE   
193   12/06/2009   Albury     -1.3      8.2       0.0         ESE   
2673  24/06/2016   Albury      6.6      8.2       4.0           W   
919   08/07/2011   Albury     -0.5      8.3       0.0         WNW   
...          ...      ...      ...      ...       ...         ...   
984   11/09/2011   Albury      NaN      NaN       NaN         NaN   
985   12/09/2011   Albury      NaN      NaN       NaN         NaN   
1529  07/05/2013   Albury      3.9      NaN       0.0         NaN   
2031  21/09/2014   Albury      3.4      NaN       0.0         NaN   
2032  22/09/2014   Albury      NaN      NaN       NaN         NaN   

      windgustspeed winddir9am winddir3pm  windspeed9am  ...  humidity3pm  \
2328           30.0          S        SSE          13.0  ...         93.0   
2675           15.0        NaN         SE           0.0  ...         79.0   
193            15.0          E        SSE           6.0  ...         72.0   
2673           48.0        WNW          W          28.0  ...         85.0   
919            28.0        NaN          W           0.0  ...         64.0   
...             ...        ...        ...           ...  ...          ...   
984             NaN        NaN        NaN           NaN  ...          NaN   
985             NaN        NaN        NaN           NaN  ...          NaN   
1529            NaN        NaN        NaN           0.0  ...          NaN   
2031            NaN        ESE        SSE           9.0  ...         43.0   
2032            NaN        NaN        NaN           NaN  ...          NaN   

      pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  \
2328       1016.6       1017.1       8.0       8.0      4.7      6.4   
2675       1023.4       1019.0       8.0       NaN      2.3      7.4   
193        1025.7       1022.2       8.0       7.0      1.9      7.9   
2673       1004.8       1007.6       8.0       6.0      7.0      7.5   
919        1026.0       1024.7       NaN       8.0      4.3      8.0   
...           ...          ...       ...       ...      ...      ...   
984           NaN          NaN       NaN       NaN      NaN      NaN   
985           NaN          NaN       NaN       NaN      NaN      NaN   
1529       1028.6          NaN       NaN       NaN      9.8      NaN   
2031       1030.6       1027.5       NaN       NaN     12.2     18.6   
2032          NaN          NaN       NaN       NaN      NaN      NaN   

      raintoday  raintomorrow difference  
2328       Rain          Rain        2.4  
2675     NoRain        NoRain        8.8  
193      NoRain        NoRain        9.5  
2673       Rain          Rain        1.6  
919      NoRain          Rain        8.8  
...         ...           ...        ...  
984         NaN           NaN        NaN  
985         NaN           NaN        NaN  
1529     NoRain           NaN        NaN  
2031     NoRain           NaN        NaN  
2032        NaN           NaN        NaN  

[3040 rows x 22 columns]

II. YOL

In [26]:
df_Albury = df[df.location == "Albury"].sort_values(by = "maxtemp", ascending=False)
df_Albury

date location  mintemp  maxtemp  rainfall windgustdir  \
68    07/02/2009   Albury     22.3     44.8       0.0         NNW   
2904  10/02/2017   Albury     23.0     43.7       0.0         WSW   
1783  16/01/2014   Albury     23.0     43.6       0.0           N   
1435  05/01/2013   Albury     19.8     43.4       0.0          NW   
62    01/02/2009   Albury     28.0     43.1       0.0          SW   
...          ...      ...      ...      ...       ...         ...   
984   11/09/2011   Albury      NaN      NaN       NaN         NaN   
985   12/09/2011   Albury      NaN      NaN       NaN         NaN   
1529  07/05/2013   Albury      3.9      NaN       0.0         NaN   
2031  21/09/2014   Albury      3.4      NaN       0.0         NaN   
2032  22/09/2014   Albury      NaN      NaN       NaN         NaN   

      windgustspeed winddir9am winddir3pm  windspeed9am  ...  humidity3pm  \
68             59.0         SE         NW           9.0  ...         11.0   
2904           44.0        SSE          W           7.0  ...         17.0   
1783           35.0          E        SSW           9.0  ...         12.0   
1435           81.0          S        ESE           7.0  ...         17.0   
62             83.0         SE          E          15.0  ...          9.0   
...             ...        ...        ...           ...  ...          ...   
984             NaN        NaN        NaN           NaN  ...          NaN   
985             NaN        NaN        NaN           NaN  ...          NaN   
1529            NaN        NaN        NaN           0.0  ...          NaN   
2031            NaN        ESE        SSE           9.0  ...         43.0   
2032            NaN        NaN        NaN           NaN  ...          NaN   

      pressure9am  pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  \
68         1008.2       1003.8       NaN       NaN     29.6     43.4   
2904       1009.9       1006.1       NaN       NaN     28.2     42.4   
1783       1016.6       1012.2       NaN       NaN     31.8     41.7   
1435       1012.4       1009.8       NaN       1.0     30.7     42.4   
62         1012.5       1008.7       NaN       NaN     32.4     41.9   
...           ...          ...       ...       ...      ...      ...   
984           NaN          NaN       NaN       NaN      NaN      NaN   
985           NaN          NaN       NaN       NaN      NaN      NaN   
1529       1028.6          NaN       NaN       NaN      9.8      NaN   
2031       1030.6       1027.5       NaN       NaN     12.2     18.6   
2032          NaN          NaN       NaN       NaN      NaN      NaN   

      raintoday  raintomorrow difference  
68       NoRain        NoRain       22.5  
2904     NoRain        NoRain       20.7  
1783     NoRain        NoRain       20.6  
1435     NoRain          Rain       23.6  
62       NoRain        NoRain       15.1  
...         ...           ...        ...  
984         NaN           NaN        NaN  
985         NaN           NaN        NaN  
1529     NoRain           NaN        NaN  
2031     NoRain           NaN        NaN  
2032        NaN           NaN        NaN  

[3040 rows x 22 columns]

16. Find out the NaN counts for each column.

In [27]:
df.isna().sum()

date                 0
location             0
mintemp           1485
maxtemp           1261
rainfall          3261
windgustdir      10326
windgustspeed    10263
winddir9am       10566
winddir3pm        4228
windspeed9am      1767
windspeed3pm      3062
humidity9am       2654
humidity3pm       4507
pressure9am      15065
pressure3pm      15028
cloud9am         55888
cloud3pm         59358
temp9am           1767
temp3pm           3609
raintoday         3261
raintomorrow      3267
difference        1881
dtype: int64

II.YOL

In [28]:
NaN = df.isna().sum().to_frame("isNaN").reset_index()
NaN

index  isNaN
0            date      0
1        location      0
2         mintemp   1485
3         maxtemp   1261
4        rainfall   3261
5     windgustdir  10326
6   windgustspeed  10263
7      winddir9am  10566
8      winddir3pm   4228
9    windspeed9am   1767
10   windspeed3pm   3062
11    humidity9am   2654
12    humidity3pm   4507
13    pressure9am  15065
14    pressure3pm  15028
15       cloud9am  55888
16       cloud3pm  59358
17        temp9am   1767
18        temp3pm   3609
19      raintoday   3261
20   raintomorrow   3267
21     difference   1881

17. Remove the rows with NaN values in "windgustdir" column from the dataframe permanently.

In [29]:
df.dropna(subset = ["windgustdir"],inplace = True)
df.head(10)

date location  mintemp  maxtemp  rainfall windgustdir  windgustspeed  \
0  01/12/2008   Albury     13.4     22.9       0.6           W           44.0   
1  02/12/2008   Albury      7.4     25.1       0.0         WNW           44.0   
2  03/12/2008   Albury     12.9     25.7       0.0         WSW           46.0   
3  04/12/2008   Albury      9.2     28.0       0.0          NE           24.0   
4  05/12/2008   Albury     17.5     32.3       1.0           W           41.0   
5  06/12/2008   Albury     14.6     29.7       0.2         WNW           56.0   
6  07/12/2008   Albury     14.3     25.0       0.0           W           50.0   
7  08/12/2008   Albury      7.7     26.7       0.0           W           35.0   
8  09/12/2008   Albury      9.7     31.9       0.0         NNW           80.0   
9  10/12/2008   Albury     13.1     30.1       1.4           W           28.0   

  winddir9am winddir3pm  windspeed9am  ...  humidity3pm  pressure9am  \
0          W        WNW          20.0  ...         22.0       1007.7   
1        NNW        WSW           4.0  ...         25.0       1010.6   
2          W        WSW          19.0  ...         30.0       1007.6   
3         SE          E          11.0  ...         16.0       1017.6   
4        ENE         NW           7.0  ...         33.0       1010.8   
5          W          W          19.0  ...         23.0       1009.2   
6         SW          W          20.0  ...         19.0       1009.6   
7        SSE          W           6.0  ...         19.0       1013.4   
8         SE         NW           7.0  ...          9.0       1008.9   
9          S        SSE          15.0  ...         27.0       1007.0   

   pressure3pm  cloud9am  cloud3pm  temp9am  temp3pm  raintoday  raintomorrow  \
0       1007.1       8.0       NaN     16.9     21.8     NoRain        NoRain   
1       1007.8       NaN       NaN     17.2     24.3     NoRain        NoRain   
2       1008.7       NaN       2.0     21.0     23.2     NoRain        NoRain   
3       1012.8       NaN       NaN     18.1     26.5     NoRain        NoRain   
4       1006.0       7.0       8.0     17.8     29.7     NoRain        NoRain   
5       1005.4       NaN       NaN     20.6     28.9     NoRain        NoRain   
6       1008.2       1.0       NaN     18.1     24.6     NoRain        NoRain   
7       1010.1       NaN       NaN     16.3     25.5     NoRain        NoRain   
8       1003.6       NaN       NaN     18.3     30.2     NoRain          Rain   
9       1005.7       NaN       NaN     20.1     28.2       Rain        NoRain   

  difference  
0        9.5  
1       17.7  
2       12.8  
3       18.8  
4       14.8  
5       15.1  
6       10.7  
7       19.0  
8       22.2  
9       17.0  

[10 rows x 22 columns]

18. Create a new dataframe, use "Location" column as the index of the dataframe, display the min, max, and median values of "evaporation" and "sunshine" columns in this dataframe.

In [ ]:
df_2 = df.copy()  
min_val = df_2.groupby("Location")[["Evaporation","Sunshine"]].min()
min_val

In [ ]:
max_val = df_2.groupby('Location')[['Evaporation','Sunshine']].max()
max_val

In [ ]:
median_value = new_df.groupby('Location')[['Evaporation','Sunshine']].median()
median_val

In [ ]:
df_min_max_median = pd.concat([min_val,max_val,median_val], axis = 1)
df_min_max_median

19. Find out the hottest day of "Perth". Example output: Timestamp('2015-01-05 00:00:00')

In [31]:
df_Albury = df[df.location == 'Perth'].sort_values(by='maxtemp', ascending=False)
df.date[df_Albury['maxtemp'].idxmax()]

'05/01/2015'

20. Group your dataframe by location and find out the averages of all numeric values.

In [32]:
df.groupby('location').mean()

mintemp    maxtemp  rainfall  windgustspeed  windspeed9am  \
location                                                                        
Adelaide          12.571487  22.891254  1.576076      36.499211      9.954804   
Albury             9.518588  22.629555  1.904552      32.977394      8.209720   
AliceSprings      13.155254  29.281325  0.869484      40.531281     14.754603   
BadgerysCreek     11.135741  24.014949  2.215406      33.600409      8.053590   
Ballarat           7.348670  18.267930  1.706369      44.982719     19.976024   
Bendigo            8.611841  21.625609  1.631586      38.844948     12.945630   
Brisbane          16.414204  26.436043  3.119424      28.212881      6.950508   
Cairns            21.225099  29.559550  5.701852      38.026150     15.902649   
Canberra           6.670100  20.731579  1.745920      40.071682     10.444228   
Cobar             13.131412  25.853488  1.132655      36.750590     14.679717   
CoffsHarbour      14.522998  23.971493  4.990985      39.227925     15.273139   
Dartmoor           8.620518  19.696769  2.150272      38.898299     10.815458   
Darwin            23.190022  32.538259  5.076454      40.579917     15.043561   
GoldCoast         17.287324  25.713189  3.737301      42.476370     19.331962   
Hobart             9.080883  17.858517  1.604739      47.530915     16.291167   
Katherine         20.546272  34.915762  3.239856      38.771465     12.100129   
Launceston         7.858822  18.983062  2.016483      35.625749      8.471547   
Melbourne         11.770359  20.760489  1.874019      45.610255     19.091281   
MelbourneAirport   9.962558  20.478958  1.456045      46.919172     20.244905   
Mildura           10.738782  24.848202  0.945370      37.168775     14.000999   
Moree             12.909625  26.975060  1.612936      39.889116     18.089170   
MountGambier       8.820797  19.818882  2.078296      42.853030     17.496651   
MountGinini        3.487122  11.506027  3.317951      46.253951     15.807078   
Nhil               8.991959  22.394576  0.933973      42.542438     16.435227   
NorahHead         15.346429  22.583152  3.388950      42.183051     13.866008   
NorfolkIsland     16.868073  21.827935  3.132717      42.601889     20.171119   
Nuriootpa          9.404508  21.751250  1.380203      40.681081     14.239513   
PearceRAAF        12.295443  26.052760  1.691525      43.579958     16.044787   
Penrith           12.537888  24.733782  2.176233      31.807330      6.345997   
Perth             12.921769  25.037402  1.905144      34.882999     10.697302   
PerthAirport      12.593723  25.541714  1.744988      44.201822     17.760297   
Portland           9.586485  18.047209  2.528629      42.059852     16.316852   
Richmond          11.333866  24.455586  2.130970      34.872290      7.016769   
Sale               8.595363  20.321583  1.523612      42.479777     14.335311   
SalmonGums         9.376693  24.288645  1.037027      39.748372     16.293269   
Sydney            15.062213  23.292758  3.477445      41.757589     15.265740   
SydneyAirport     14.893845  23.394050  2.984631      47.234280     18.559689   
Townsville        20.412931  29.361485  3.455083      38.840570     15.233831   
Tuggeranong        7.235581  20.771295  2.165203      35.215883      7.641560   
Uluru             14.399533  30.389633  0.714555      41.268373     17.610928   
WaggaWagga         9.628212  23.046863  1.711737      36.648105     12.881248   
Walpole           11.837917  20.491127  2.865308      39.829640     13.924143   
Watsonia          10.115729  20.858445  1.842159      38.059219      9.922585   
Williamtown       12.774443  23.985346  3.621076      41.681518     16.419861   
Witchcliffe       10.786932  21.685407  2.910366      40.083025     13.959946   
Wollongong        14.919289  21.445166  3.611757      45.690842     16.637222   
Woomera           13.410819  26.639327  0.496816      44.156566     20.098720   

                  windspeed3pm  humidity9am  humid